app/javascript/dashboard/components-next/filter/provider.js

```diff
@@ -0,0 +1,299 @@
+ * @property {'multiSelect'|'searchSelect'|'plainText'|'date'|'booleanSelect'} input_type - The input type for the attribute
+ * @property {FilterOption[]} [options] - The options available for the attribute if it is a multiSelect or singleSelect type
+ * @property {'text'|'number'} data_type
+ * @property {FilterOperator[]} filter_operators - The operators available for the attribute
+ * @property {'standard'|'additional'|'customAttributes'} attribute_model
+ */
+
+/**
+ * @typedef {Object} FilterGroup
+ * @property {string} name
+ * @property {FilterType[]} attributes
+ */
+
+/**
+ * Determines the input type for a custom attribute based on its key
+ * @param {string} key - The attribute display type key
+ * @returns {'date'|'plainText'|'searchSelect'|'booleanSelect'} The corresponding input type
+ */
+const customAttributeInputType = key => {
+  switch (key) {
+    case 'date':
+      return 'date';
+    case 'text':
+      return 'plainText';
+    case 'list':
+      return 'searchSelect';
+    case 'checkbox':
+      return 'booleanSelect';
+    default:
+      return 'plainText';
+  }
+};
+
+/**
+ * Composable that provides conversation filtering context
+ * @returns {{ filterTypes: import('vue').ComputedRef<FilterType[]>, fitlerGroups: import('vue').ComputedRef<FilterGroup[]> }}
+ */
+export function useConversationFilterContext() {
+  const { t } = useI18n();
+
+  const conversationAttributes = useMapGetter(
+    'attributes/getConversationAttributes'
+  );
+
+  const labels = useMapGetter('labels/getLabels');
+  const agents = useMapGetter('agents/getAgents');
+  const inboxes = useMapGetter('inboxes/getInboxes');
+  const teams = useMapGetter('teams/getTeams');
+  const campaigns = useMapGetter('campaigns/getAllCampaigns');
+
+  const {
+    equalityOperators,
+    presenceOperators,
+    containmentOperators,
+    dateOperators,
+    getOperatorTypes,
+  } = useOperators();
+
+  /**
+   * @type {import('vue').ComputedRef<FilterType[]>}
+   */
+  const customFilterTypes = computed(() => {
+    return conversationAttributes.value.map(attr => {
+      return {
+        attribute_key: attr.attributeKey,
+        value: attr.attributeKey,
+        attribute_name: attr.attributeDisplayName,
+        label: attr.attributeDisplayName,
+        input_type: customAttributeInputType(attr.attributeDisplayType),
+        filter_operators: getOperatorTypes(attr.attributeDisplayType),
+        options:
+          attr.attributeDisplayType === 'list'
+            ? attr.attributeValues.map(item => ({ id: item, name: item }))
+            : [],
+        attribute_model: 'customAttributes',
+      };
+    });
+  });
+
+  /**
+   * @type {import('vue').ComputedRef<FilterType[]>}
+   */
+  const filterTypes = computed(() => [
+    {
+      attribute_key: 'status',
+      value: 'status',
+      attribute_name: t('FILTER.ATTRIBUTES.STATUS'),
+      label: t('FILTER.ATTRIBUTES.STATUS'),
+      input_type: 'multiSelect',
```

Comment:
status is implemented as single select in the backend. This is why there is an `all` key.